### DISCUSSION TITLE
Powershell Windows Output stops after ~100 lines

### DISCUSSION THREAD

**Discussion by MikeDombo** [community/community#25333] [Actions]
<p>Hi Folks,<br>
I’m trying a cross-platform build with Maven and it works great on linux, but on Windows it stops outputting anything to stdout or stderr after about 100 lines; right when the JUnit tests begin.</p>
<p>Any thoughts or similar experiences?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mikedombo">@mikedombo</a>,</p>
<p>Thank you for being here!<br>
Is there any error reported? Did you try to pipe the test result and then for check? Could you please share your repo/sample repo for further investigation?</p>
<p>Thanks</p>

**Comment by MikeDombo** [comment]
<p>This is the end of the output that we get. Just before this output is the beginning of our test run output, we’d expect many many more lines, so powershell is definitely hiding some output. Our tests fail, which is fine, so that’s why powershell shows the exit code of 1.</p>
<pre><code class="lang-auto">mvn : Threads are still running: [surefire-forkedjvm-command-thread, junit-jupiter-timeout-watcher]
At C:\actions-runner\_work\_temp\906e8779-c2d0-4cad-98f4-c18f1230df36.ps1:2 char:1
+ mvn -ntp -U clean verify 2&gt;&amp;1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Threads are sti...imeout-watcher]:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
##[error]Process completed with exit code 1.
</code></pre>
<p>Sorry, but I cannot share the repository with you. I’ll see if I can get an example that has the same issue.</p>
<p>EDIT:<br>
I tried to just redirect the output to a file, but even the file was missing most of the output. This is very strange indeed because when we run the same maven command interactively, we do see all of the output. So it seems like there’s somethin different either in powershell or maven when not running interactively.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mikedombo">@mikedombo</a>,</p>
<p>Thanks for your reply!<br>
This error could be caused by that command doesn’t exit properly. Please add <code>&amp; exit 0</code> with your script or command. Please refer to the similar <a href="https://github.community/t/unable-to-use-binary-in-github-workflow/116256/2">ticket</a> for your reference.<br>
Besides, if the workflow logs do not provide enough detail to diagnose why a workflow, job, or step is not working as expected, you can enable additional debug logging. Please check doc <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run#enabling-debug-logging" rel="nofollow noopener">here</a> for more details.</p>
<p>Thanks.</p>

**Comment by MikeDombo** [comment]
<p>Sorry, but that’s not correct. Exiting with a failure is correct because the build failed.</p>
<p>The timeline is<br>
Start maven<br>
Get 100 lines of output<br>
Junit surefire tests start<br>
Get a couple more lines of output<br>
10 minutes later, when the tests have been outputting thousands of lines which don’t appear, then maven exits 1 because the build failed.</p>
<p>We should have tons of logs, the same that we get on Linux, during the 10 minutes of testing. However, nothing shows up until the error message that it exited non-zero, which again is expected.</p>
<p>Running the same command in powershell on the same host works as expected and we get all the standard output</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mikedombo">@mikedombo</a>,</p>
<p><a href="https://github.com/actions/runner/blob/11435857e4cd9795fa5ee68c20b674e5f96052ab/src/Runner.Worker/Handlers/ScriptHandlerHelpers.cs#L57" rel="nofollow noopener">The runner prepends  <code>$erroractionpreference='stop'</code> </a> which may halt execution. Could you please add <code>$erroractionpreference = 'continue'</code> to your script for a try?<br>
Add <code>&amp; exit 0</code> to the command will not affact the test result, but help to return an exit code to runner.<br>
Please also try to enable debug logging, and if possible provide more error details for further investigation.</p>
<p>Thanks.</p>

**Comment by MikeDombo** [comment]
<p>Adding <code>&amp; exit 0</code> doesn’t work due to</p>
<pre><code class="lang-auto">The ampersand (&amp;) character is not allowed. The &amp; operator is reserved for future use; wrap an ampersand in double 
quotation marks ("&amp;") to pass it as part of a string.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : AmpersandNotAllowed
</code></pre>

**Comment by MikeDombo** [comment, marked as answer]
<p>The answer is related to <code>$erroractionpreference</code>. It would seem that Powershell is reading our stderr and, when it sees the first line from that it throws the <code>NativeCommandError</code>, then it stops outputting all of our stdout and stderr. The command continue to execute all the tests, but eventually fails.</p>
<p>There are some weird ways to disable this behavior on powershell, but I think the best solution for us is to just fallback to cmd instead of powershell.</p>
